src/Mvc/src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageActionDescriptorProvider.cs

```diff
@@ -106,6 +107,10 @@ private void AddActionDescriptors(IList<ActionDescriptor> actions, PageRouteMode
                     descriptor.RouteValues.Add("page", model.ViewEnginePath);
                 }
 
+                // Mark all pages as a "dynamic endpoint" - this is how we deal with the compilation of pages
+                // in endpoint routing.
+                descriptor.EndpointMetadata.Add(new DynamicEndpointMetadata());
```

Comment:
Is this tested for and used anywhere? ContainsDynamicEndpoints uses it but ContainsDynamicEndpoints doesn't appear to be used in any runtime code.

Reply:
I guess PageLoaderMatcherPolicy is making the assumption that anything with PageActionDescriptor is implicitly dynamic.

Reply:
I added it to make you guys happy. This is here so **other** policies can know that we have dynamic behaviour. Right now there are no such other policies.